sphinx/config.py

```diff
@@ -213,7 +213,7 @@ class Config:
         'figure_language_filename': _Opt('{root}.{language}{ext}', 'env', frozenset((str,))),
         'gettext_allow_fuzzy_translations': _Opt(False, 'gettext', ()),
         'translation_progress_classes': _Opt(
-            False, 'env', ENUM(True, False, 'translated', 'untranslated')),
+            False, 'env', ENUM(True, False, '1', '0', 'translated', 'untranslated')),
```

Comment:
Wouldn't it be better to coerce the type when processing the `define` parameter? By adding '0', '1' here, we (i) allow `translation_progress_classes = '0'` in `conf.py`, which seems unnecessarily permissive, and (ii) more importantly, every usage site has to check both truthy variants. - That's still okish in the present case, because there are only two usages, but it's not a pattern that scales for more heavily used parameters.

Reply:
I more-or-less agree, but I get the strange sense that there could be unintended consequences there, because information entered at the command-line is, essentially, always text/`str`-typed.

So to try to reason a bit more about it before moving ahead: would that imply that any `'0'` or `'1'` value received as a `define` value should always be coerced to `bool`?

Reply:
I think iff `bool` is a valid type for the config value, then `0` and `1` should be converted. I agree with Tim that this should ideally be done at the command-line-parsing stage, as we offer `-D spam=0` as a shortcut to `spam = False`.

A

Reply:
If we extend the parsing, it's better to have a utillity function, namely that says which kind of objects are to be considered "yes" and which are to be considered as "no".

Reply:
Should the empty-string be valid as a value for command-line `define` arguments?

Reply:
I don't really want to accept the empty string to be a valid value. Because I'm not sure you can distinguish between `-DATTR=` from `-DATTR=""` and in the first case, I see it more as "use the default value of ATTR" or "do not set ATTR".

Since it's ambiguous, I'd prefer not to handle it for now.